lib/liquid/tags/include.rb

```diff
@@ -70,9 +70,11 @@ def render_to_output_buffer(context, output)
 
       old_template_name = context.template_name
       old_partial       = context.partial
+
       begin
-        context.template_name = template_name
-        context.partial       = true
+        context.template_name = partial.name if partial.name
```

Comment:
I think this can be set unconditionally.  The reason why I was saying to make it conditional in Template#render (if we support a template name parse argument) is to preserve the behaviour of any application that sets Context#template_name on the passed in Context.

```suggestion
        context.template_name = partial.name
```